run.py

```diff
@@ -0,0 +1,307 @@
+
+import sys
+import re
+import argparse
+import os
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.sql import select
+
+from adsdata import row_view
+from adsdata import metrics
+from adsdata import reader
+from adsputils import load_config, setup_logging
+from adsmsg import NonBibRecord
+
+from adsdata.tasks import task_output_results
+
+logger = None
+config = {}
+
+
+def load_column_files(config, sql_sync):
+    """ use psycopg.copy_from to data from column file to postgres
+    
+    after data has been loaded, join to create a unified row view 
+    """
+    conn = sql_sync.engine.raw_connection()
+    cur = conn.cursor()
+    
+    for t in row_view.SqlSync.all_types:
+        table_name = sql_sync.schema + '.' + t
+        logger.info('processing {}'.format(table_name))
+        filename = config['DATA_PATH'] + config[t.upper()]
+        if t == 'canonical':
+            r = reader.BibcodeFileReader(filename)
+        elif t == 'refereed':
+            r = reader.RefereedFileReader(filename) 
+        else:
+            r = reader.StandardFileReader(t, filename)
+        if r:
+            cur.copy_from(r, table_name)
+            conn.commit()
+
+    cur.close()
+    conn.close()
+    sql_sync.create_joined_rows()
+
+def load_metrics(m, row_view_schema):
+    """
+    """
+    Session = sessionmaker()
+    sess = Session(bind=m.connection)
+    command_args = ' populateMetricsTable '  \
+                   + ' --rowViewSchemaName ' + row_view_schema \
+                   + ' --metricsSchemaName ' + m.schema
+    python_command = "'python " + os.path.abspath(__file__) + command_args + "'"
+    sql_command = 'copy ' + m.schema + '.metrics'   \
+                  + ' from program ' + python_command + ';'
+    
+    sess.execute(sql_command)
+    sess.commit()
+    sess.close()
+
+def nonbib_to_master_pipeline(schema, batch_size=1):
+    """send all nonbib data to queue for delivery to master pipeline"""
+    nonbib = row_view.SqlSync(schema)
+    connection = nonbib.engine.connect()
+    s = select([nonbib.table])
+    results = connection.execute(s)
+    recs = []
+    for current_row in results:
+        current_row = dict(current_row)
+        logger.debug('Will forward this record: %s', current_row)
+        rec = NonBibRecord(**current_row)
+        recs.append(rec)
+        if len(recs) >= batch_size:
+            if len(recs) == 1:
+                logger.debug("Calling 'app.forward_message' with '%s'", str(rec))
+                task_output_results.delay(rec)
+            else:
+                logger.debug("Calling 'app.forward_message' with '%s'", str(recs))
+                task_output_results.delay(recs)
```

Comment:
Depending on this if, task_output_results will receive a list of NonBibRecords or a single NonBibRecords. We will have to control this in master or agree to always send lists (even of just 1 element).